charts/dapr/charts/dapr_rbac/templates/ClusterRoleBinding.yaml

```diff
@@ -29,46 +29,138 @@ apiVersion: rbac.authorization.k8s.io/v1
-  verbs: ["get"]
-- apiGroups: ["*"]
-  resources: ["deployments", "statefulsets", "services", "components", "configurations", "subscriptions", "leases", "secrets"]
-  verbs: ["list"]
-- apiGroups: ["*"]
-  resources: ["deployments", "statefulsets", "services", "components", "configurations", "subscriptions", "leases", "secrets"]
-  verbs: ["watch"]
-- apiGroups: ["*"]
-  resources: ["services", "secrets", "subscriptions", "configmaps", "leases", "services/finalizers", "deployments/finalizers", "statefulsets/finalizers"]
-  verbs: ["update"]
-- apiGroups: ["*"]
-  resources: ["customresourcedefinitions"]
-  verbs: ["patch"]
-- apiGroups: ["*"]
-  resources: ["services", "leases"]
-  verbs: ["delete"]
-- apiGroups: ["*"]
-  resources: ["deployments", "statefulsets", "services", "configmaps", "events", "leases"]
-  verbs: ["create"]
+  - apiGroups:
+      - apps
+    resources:
+      - statefulsets
+    verbs:
+      - list
+      - get
+      - update
+      - create
+      - delete
+      - watch
+
+  - apiGroups:
+      - apiextensions.k8s.io
+    resources:
+      - customresourcedefinitions
+    verbs:
+      - list
+      - get
+      - update
+      - create
+      - delete
+
+  - apiGroups:
+      - ""
+    resources:
+      - serviceaccounts
+      - services
+      - configmaps
+    verbs:
+      - get
+      - list
+      - update
+      - create
+
+  - apiGroups:
+      - coordination.k8s.io
+    resources:
+      - leases
+    verbs:
+      - get
+      - list
+      - update
+      - create
+  - apiGroups:
+      - apps
+    resources:
+      - deployments
+    verbs:
+      - get
+  - apiGroups:
+      - dapr.io
+    resources:
+      - components
+      - configurations
+    verbs:
+      - get
+
+  - apiGroups:
+      - dapr.io
+    resources:
+      - components
+      - components/status
+      - components/scale
+      - components/finalizers
+    verbs:
+      - create
+      - get
+      - list
+      - update
+      - patch
+      - watch
+      - delete
+      - deletecollection
+
+  - apiGroups:
+      - dapr.io
+    resources:
+      - configurations
+      - configurations/status
+      - configurations/scale
+      - configurations/finalizers
+    verbs:
+      - create
+      - get
+      - list
+      - update
+      - patch
+      - watch
+      - delete
+      - deletecollection
+
+  - apiGroups:
+      - dapr.io
+    resources:
+      - subscriptions
+      - subscriptions/status
+      - subscriptions/scale
+      - subscriptions/finalizers
+    verbs:
+      - create
+      - get
+      - list
+      - update
+      - patch
+      - watch
+      - delete
+      - deletecollection
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
   name: secret-reader
-  namespace: default
+  namespace: {{ .Release.Namespace }}
```

Comment:
This actually needs to be in default.

Reply:
Can you explain why this needs to be in default namespace ? 
For example:  My app+side car in namespace=myapp123 and dapr controlplane namesapce=dapr-system. Why we need this in default namespace. Is default is meant to be the one that the side car deployed in  ?(allowing side car to access secrets) ?

Reply:
These roles are intended for the Secrets API specifically, and this is why they are set up automatically on the `default` namespace which is where apps (deployments) are running by default on Kubernetes. This role is not intended for the Dapr control plane and thus should not be in the `Release.Namespace`.